packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -1479,14 +1520,22 @@ export async function wrapGetDeploymentError(
     deployCtx.ctx.logProvider?.error(
       getLocalizedString("core.deployArmTemplates.FailedToGetDeploymentError", error.message)
     );
-    const returnError = new Error(
-      getLocalizedString(
-        "core.deployArmTemplates.FailedToGetDeploymentErrorNotification",
-        deployCtx.deploymentName,
-        deployCtx.resourceGroupName
+    return err(
+      new UserError(
+        SolutionSource,
+        "GetDeploymentErrorFailed",
+        getDefaultString(
```

Comment:
why there need defaultString and localizedstring?

Reply:
getDefaultString() is default English version used for telemetry.
getLocalizedString() is localized version used for UI display.